### PR TITLE
Feature/bai 464 email deployment owner on model version deletion

### PR DESCRIPTION
**Pull request by JR40159**

### CODE REVIEW

File: .vscode/launch.json
Comment (ARADDCC012) [inline .vscode/launch.json:2]: You can't put comments in JSON

File: server/services/model.ts
Comment (ARADDCC012) [inline server/services/model.ts:108]: I don't think this comment is correct. It's removing the version of a model, not the model itself.

I'd also argue that the comment isn't needed as the function name describes this clearly, but I don't feel too strongly about that.

File: server/services/deployment.ts
Comment (ARADDCC012) [inline server/services/deployment.ts:75]: Was this removed for being dead/unused code?

Comment (JR40159) [reply server/services/deployment.ts:75]: Yes, I couldn't find any references to it and I don't think we include version info in deployment records anymore

File: .vscode/launch.json
Comment (JR40159) [reply .vscode/launch.json:2]: Those comments were auto-generated but I can remove them

File: server/services/deployment.spec.ts
Comment (ARADDCC012) [inline server/services/deployment.spec.ts:82]: Instead of making new test data here you could construct the array of users by using the existing test data in `utils/test/testModels`. E.g. `testUser` and `testUser2`.

File: server/services/deployment.spec.ts
Comment (ARADDCC012) [inline server/services/deployment.spec.ts:80]: Unfortunately our test data isn't fully representative of the full DeploymentDoc and won't have all the properties of a full document. It's probably best to leave it to infer the type itself at the moment, which will just be `any[]` as we haven't typed our test data.

File: server/services/deployment.spec.ts
Comment (ARADDCC012) [inline server/services/deployment.spec.ts:90]: Typo `send` -> `sent`

File: server/routes/v1/version.ts
Comment (ARADDCC012) [inline server/routes/v1/version.ts:346]: I would think that we should expect to find the model 100% of the time so it would be a valid reason to throw an error if we don't:
```
if (!model) {
  throw NotFound({ code: 'model_not_found', version.model }, `Unable to find model '${version.model}'`)
}

const deployments = await findDeploymentsByModel(user, model)
if (deployments.length > 0) {
  emailDeploymentOwnersOnVersionDeletion(deployments, version)
}
```

File: server/routes/v1/version.spec.ts
Comment (ARADDCC012) [inline server/routes/v1/version.spec.ts:21]: Could be written as `removeVersionFromModel: jest.fn(async () => undefined)`

File: server/routes/v1/version.spec.ts
Comment (ARADDCC012) [inline server/routes/v1/version.spec.ts:28]: You can just make this a jest function for now as you're setting the return values in your tests (which is the right way to do it anyway) so these values will be overwritten at that point. `findDeploymentsByModel: jest.fn()`

The same applies to the other functions you're mocking in this file. We only tend to set a specific return value at this level is we want the same return value for all tests in this file. Otherwise just setting `jest.fn()` here and using `mockReturnValueOnce` or `mockImplementationOnce` in each test that uses them is the preferred approach.

File: server/routes/v1/version.spec.ts
Comment (ARADDCC012) [inline server/routes/v1/version.spec.ts:58]: Doesn't really matter but you could instead use `mockReturnValueOnce(['deployment'])`. `mockImplementation` is more useful if you want to change the functionality of something in a test (which is pretty uncommon)

File: server/services/deployment.spec.ts
Comment (ARADDCC012) [reply server/services/deployment.spec.ts:23]: My mistake, I did check this but got it wrong

File: .vscode/launch.json
Comment (a3957273) [reply .vscode/launch.json]: VS: Code always using 'JSONC', which allows comments and trailing commas.  Can keep the comments in if they're useful.

File: server/routes/v1/version.spec.ts
Comment (JR40159) [reply server/routes/v1/version.spec.ts:28]: After discussion, decided to leave as is to avoid repeated code in multiple tests

**Review by ARADDCC012 — APPROVED** [review]
Approved but needs to wait for the ESM work to be merged in before this can be merged

**Pull request closed without merging**
